Test/QuickCheck/Arbitrary.hs

```diff
@@ -1,15 +1,23 @@
 -- | Type classes for random generation of values.
 {-# LANGUAGE CPP #-}
 #ifndef NO_GENERICS
-{-# LANGUAGE DefaultSignatures, FlexibleContexts, TypeOperators #-}
-{-# LANGUAGE FlexibleInstances, KindSignatures, ScopedTypeVariables #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE MagicHash #-}
 {-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+#if __GLASGOW_HASKELL__ >= 702
+{-# LANGUAGE DefaultSignatures #-}
+#endif
 #if __GLASGOW_HASKELL__ < 710
 {-# LANGUAGE OverlappingInstances  #-}
 #endif
 #endif
 #ifndef NO_SAFE_HASKELL
-{-# LANGUAGE Safe #-}
+{-# LANGUAGE Trustworthy #-}
```

Comment:
Sadly, in GHC 7.2, `GHC.Generics` was inferred as `Unsafe`, so importing it destroys safety. You could probably add some more CPP guards to get it to be safe on GHC 7.4 and later, if you so desired.